### PR TITLE
ARROW-3711: [C++] Don't pass CXX_FLAGS to C_FLAGS

### PR DESCRIPTION
**Pull request by xhochy**

### CODE REVIEW

**Comment by kou** [comment]
I'll rebase this on master to apply RAT fix.

**Comment by xhochy** [comment]
Build failures are mainly due to https://github.com/apache/arrow/pull/2909; will wait and rebase once that is merged.

File: cpp/CMakeLists.txt [hunk 1]
```diff
@@ -479,8 +480,8 @@ include(ThirdpartyToolchain)
 set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} ${CXX_COMMON_FLAGS}")
 set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} ${ARROW_CXXFLAGS}")
 
-# For any C code, use the same flags.
-set(CMAKE_C_FLAGS "${CMAKE_CXX_FLAGS}")
+# For any C code, use the same flags. These flags don't contaon
```
Comment (pitrou) [inline cpp/CMakeLists.txt]: Looks like the sentence is unfinished (and there's a typo).

**Comment by xhochy** [comment]
Travis was green, the status was thus not updated.

**Review by xhochy — COMMENTED** [review]
+1, merging.

**Pull request closed without merging**